src/app/introduction/activity-3/activity-3.component.ts

```diff
@@ -102,6 +108,10 @@ export class PicturePuzzleComponent implements OnInit {
         this._dashboardService.getProgressStatus().subscribe(response => {
             this.completed = this._sharedData.checkProgress(1, 3, response);
         });
+        this._route.params.subscribe( params => {
+          this.PUZZLE_DIFFICULTY = this.difficultyList[ params.level - 1];
```

Comment:
spacing here.

Reply:
This space is intentional as it has braces before it wrapping the code that depends on the parameter of the previous line.

Reply:
I meant this [ Params.level -1] should be [Params.level - 1]